libs/feedback/src/lib/feedback-widget/feedback-widget.component.ts

```diff
@@ -1,50 +1,73 @@
-import { ActivatedRoute } from '@angular/router';
-import { Component, HostListener, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Component, OnInit } from '@angular/core';
 import { FeedbackService } from '../feedback.service';
 import { Message } from '../message';
 import { Observable } from 'rxjs';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { debounceTime } from 'rxjs/operators';
 
-function findMatchingDOMAncestor(element) {
-  while (element.parentNode) {
-    if (
-      element.className &&
-      element.className.indexOf('feedback-container') >= 0
-    ) {
-      return true;
-    }
-    element = element.parentNode;
-  }
-}
 
 @Component({
   selector: 'feedback-widget',
   templateUrl: './feedback-widget.component.html',
   styleUrls: ['./feedback-widget.component.css']
 })
 export class FeedbackWidgetComponent implements OnInit {
+
   messages$: Observable<Message[]>;
-  open: boolean;
+
+  formGroup: FormGroup;
+  statusMessage = '';
+  error = false;
 
   constructor(
+    private fb: FormBuilder,
+    private activatedRoute: ActivatedRoute,
     private feedbackService: FeedbackService,
-    private activatedRoute: ActivatedRoute
-  ) {}
+    private router: Router
+  ) {
+    this.router.events.subscribe(() => {
```

Comment:
missing unsubscribe to dispose subscription

Reply:
![image](https://user-images.githubusercontent.com/40883170/52914909-ddfd3c00-32de-11e9-90a2-afbd63d92c51.png)
![image](https://user-images.githubusercontent.com/40883170/52914915-e190c300-32de-11e9-84f1-61e3d68a60dc.png)
![image](https://user-images.githubusercontent.com/40883170/52914917-e6557700-32de-11e9-97b8-9309bc5d85cc.png)

Like above, is it right way?